WalletWasabi.Gui/Converters/TimeSpanStringConverter.cs

```diff
@@ -0,0 +1,71 @@
+using Avalonia.Data.Converters;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+
+namespace WalletWasabi.Gui.Converters
+{
+	public class TimeSpanStringConverter : IValueConverter
+	{
+		public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+		{
+			if (value is TimeSpan ts)
+			{
+				var builder = new StringBuilder();
+				if (ts.Days != 0)
+				{
+					if (ts.Days == 1)
+					{
+						builder.Append($"{ts.Days} day ");
+					}
+					else
+					{
+						builder.Append($"{ts.Days} days ");
+					}
+				}
+				if (ts.Hours != 0)
+				{
+					if (ts.Hours == 1)
+					{
+						builder.Append($"{ts.Hours} hour ");
+					}
+					else
+					{
+						builder.Append($"{ts.Hours} hours ");
+					}
+				}
+				if (ts.Minutes != 0)
+				{
+					if (ts.Minutes == 1)
+					{
+						builder.Append($"{ts.Minutes} minute ");
+					}
+					else
+					{
+						builder.Append($"{ts.Minutes} minutes ");
+					}
+				}
+				if (ts.Seconds != 0)
+				{
+					if (ts.Seconds == 1)
+					{
+						builder.Append($"{ts.Seconds} second");
+					}
+					else
+					{
+						builder.Append($"{ts.Seconds} seconds");
+					}
+				}
+				return builder.ToString();
+			}
+
+			throw new InvalidOperationException();
```

Comment:
I would suggest to throw an `ArgumentException` the a message: "An TimeSpan was expected but a {object.GetType().Name} was received", or something like that.

Reply:
Sure, here you go: https://github.com/zkSNACKs/WalletWasabi/pull/1674/commits/cb3585c35425928bb89fac269a9d814714e5725b

Can you review it?